addon/mixins/in-viewport.js

```diff
@@ -121,16 +121,21 @@ export default Mixin.create({
       const boundingClientRect = element.getBoundingClientRect();
 
       if (boundingClientRect) {
-        this._triggerDidAccessViewport(
-          isInViewport(
-            boundingClientRect,
-            height,
-            width,
-            get(this, 'viewportTolerance')
-          )
+        let inView = isInViewport(
+          boundingClientRect,
+          height,
+          width,
+          get(this, 'viewportTolerance')
         );
-        if (get(this, 'viewportUseRAF')) {
-          rAFIDS[get(this, 'elementId')] = window.requestAnimationFrame(
+        this._triggerDidAccessViewport(inView);
+
+        if (!inView && get(this, 'viewportUseRAF')) {
```

Comment:
I think `!inView` will break the recursive calls to `_setInViewportEntered` once it comes in view.  Then it can't recover to be called again, say for an infinity list of items

Reply:
I'm seeing this called for each element on the page. Once an element `isInViewport` it should no longer need to recurse since `_triggerDidAccessViewport` will trigger the rendered change.

Reply:
I'm happy to remove the `!inView` change if that causes a problem.  I mainly wanted to cancel all outstanding requests.